scripts/consensus_merge.py

```diff
@@ -0,0 +1,587 @@
+    """ Remove leading 'chr', if any, from chrom_name and return """
+    if chrom_name.startswith('chr'):
+        return chrom_name[3:]
+    return chrom_name
+
+def stringify(iterable, sep_char=','):
+    """ Turn an iterable into a delimited string with no spaces 
+        delimiting character is ',' by default
+    """
+    return sep_char.join([str(i) for i in iterable])
+
+class Sample(object):
+    """ Object to calculate and store FORMAT information
+           for samples associated with a VCF record (Variant)
+    """
+    @staticmethod
+    def AF_from_AD(AD):
+        """ Calculate allele frequency from AD FORMAT field value
+            Note that this is done on the presumption of a 0/1 genotype
+            If this cannot be assumed (i.e. may be 1/0, 1/2) in normalized VCFs,
+                this method needs to be fixed
+        """
+        ref_counts, alt_counts = AD
+        total_counts = ref_counts + alt_counts
+        if not total_counts:
+            return 0.0
+        return alt_counts / total_counts
+
+    @staticmethod
+    def name_to_gt(name):
+        """ Convert names like 'het' to canonical genotypes """
+        if name == 'hom':
+            return '1/1'
+        elif name == 'het':
+            return '0/1'
+        elif name in ('ref', 'conflict'):
+            return '0/0'
+        else:
+            raise ValueError('No genotype associated with %s' % name)
+
+    def __init__(self, name, stype, record, caller):
+        """ Initialize this object and calculate 
+            Args:
+                name (str): the name of the sample from the VCF
+                stype (str): sample may be 'tumor' or 'normal'
+                record (pysam.VariantRecord): record with which this
+                sample is associated
+                caller (str): name of caller for this variant, 
+                    must be a member of CALLER_NAMES (case insensitive)
+
+            Raises:
+                ValueError if stype is not 'tumor' or 'normal'
+                IOError if name is not associated with exactly one
+                    sample in record
+                ValueError if caller is not in CALLER_NAMES
+        """
+        self.name = name
+        if stype not in ('tumor', 'normal'):
+            raise ValueError('Sample type must be tumor or normal, not %s' % stype)
+        self.type = stype
+        self.record = record
+
+        target_sample = [self.record.samples[index] for index in range(len(self.record.samples)) 
+                         if self.record.samples[index].name == self.name]
+        if len(target_sample) != 1:
+            raise IOError('Sample name %s not found or duplicated' % self.name)
+        self.vcf_sample = target_sample[0]
+
+        caller_test = [c.lower() for c in CALLER_NAMES]
+        if caller.lower() not in caller_test:
+            raise ValueError('Unknown caller %s' % caller)
+
+        self.caller = caller.lower()
+
+    @functools.cached_property
```

Comment:
When I try to run with no args (using python3 since python 2 is no longer supported), I get the error:
```
  File "/home/ubuntu/tools/kf-somatic-workflow/scripts/consensus_merge.py", line 47, in <module>
    class Sample(object):
  File "/home/ubuntu/tools/kf-somatic-workflow/scripts/consensus_merge.py", line 110, in Sample
    @functools.cached_property
AttributeError: module 'functools' has no attribute 'cached_property'
```

Reply:
As mentioned in the 'decorator' comment above I believe this is a Python 3.8 feature. The functionality existed before but had a different name.